src/System.Net.Http/src/System/Net/Http/Headers/HeaderDescriptor.cs

```diff
@@ -31,6 +31,7 @@ private HeaderDescriptor(string headerName)
         public string Name => _headerName;
         public HttpHeaderParser Parser => _knownHeader?.Parser;
         public HttpHeaderType HeaderType => _knownHeader == null ? HttpHeaderType.Custom : _knownHeader.HeaderType;
+        public KnownHeader KnownHeader => _knownHeader;
```

Comment:
The general pattern in this class is to not expose this directly, but instead have methods that use this internally and provide the necessary functionality.  See for example TryGet below.  So perhaps we should expose WriteBytesAsync or something here.

I can't remember if there was a specific reason for this or not; I think there was at one point, but it may no longer matter.

Reply:
> So perhaps we should expose WriteBytesAsync or something here.

That would require teaching HeaderDescriptor, which is general header infrastructure, about the specifics of SocketsHttpHandler, effectively giving HeaderDescriptor knowledge of HttpConnection.  I don't think we should do that.  Alternatively we could introduce an abstraction, but that would add cost and complexity.  I'm not seeing the downside to exposing KnownHeader in this way; this is all internal anyway.

Reply:
I agree we shouldn't make HeaderDescriptor aware of the HttpConnection stuff.  I'm fine just exposing KnownHeader here.